dds/idl/be_global.cpp

```diff
@@ -606,3 +655,39 @@ BE_GlobalData::has_key(AST_Union* node)
   // Check for @key on the discriminator
   return node && node->disc_annotations().find(key_annotation_);
 }
+  
+/**
+ * @brief checks to see if we are of type nested
+ * @param node the node to investigate
+ * @return if we are a nested node or not
+ * @author ceneblock
+ */
+bool
+BE_GlobalData::is_nested_type(AST_Decl* node)
+{
+  ///foolishly assuming that if we will call both in sucession.
+  bool isTopic = is_topic_type(node);
+
+  bool rv = false;
+  if (node) {
+    if (node->node_type() == AST_Decl::NT_struct) {
+      rv = node->annotations().find(nested_annotation_);
```

Comment:
You will actually have to check the `value` of nested.

Reply:
Won't this mean `is_topic_type` will need to be changed as well?

Reply:
For now I don't think we will be supporting the members of `@topic`, mostly because the OMG doesn't say much about them. For now I think it's enough that we have the standard definition so that opendds_idl doesn't error when someone tries to use the arguments of `@topic`.